wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -91,6 +102,12 @@ def _is_container_creating(status: "V1PodStatus") -> bool:
     return False
 
 
+def _iso8601_to_timestamp(iso8601_str: str) -> float:
+    """Convert an ISO8601 string to a timestamp."""
+    dt = datetime.fromisoformat(iso8601_str.replace("Z", "+00:00"))
+    return dt.replace(tzinfo=timezone.utc).timestamp()
```

Comment:
i'm not sure about this `.replace()` .. seems like it would break for iso8601 inputs that have a non-UTC ("Z") timezone

Reply:
Is there not a way to do this using datetime or dateutil packages?

Reply:
yep this was bad, I am just using dateutil now

Reply:
Also added a whole s u i t e of tests